src/base/bittorrent/sessionimpl.cpp

```diff
@@ -4350,23 +4351,52 @@ void SessionImpl::resume()
     }
 }
 
-int SessionImpl::maxConnectionsPerTorrent() const
+int SessionImpl::maxConnectionsPerDownloadingTorrent() const
 {
-    return m_maxConnectionsPerTorrent;
+    return m_maxConnectionsPerDownloadingTorrent;
 }
 
-void SessionImpl::setMaxConnectionsPerTorrent(int max)
+int SessionImpl::maxConnectionsPerSeedingTorrent() const
+{
+    return m_maxConnectionsPerSeedingTorrent;
+}
+
+void SessionImpl::setMaxConnectionsPerDownloadingTorrent(int max)
 {
     max = (max > 0) ? max : -1;
-    if (max != maxConnectionsPerTorrent())
+    if (max != maxConnectionsPerDownloadingTorrent())
     {
-        m_maxConnectionsPerTorrent = max;
+        m_maxConnectionsPerDownloadingTorrent = max;
 
         for (const TorrentImpl *torrent : asConst(m_torrents))
         {
             try
             {
-                torrent->nativeHandle().set_max_connections(max);
+                if (m_maxConnectionsPerSeedingTorrent == -1 || !torrent->isUploading())
```

Comment:
```suggestion
                if ((m_maxConnectionsPerSeedingTorrent == -1) || !torrent->isUploading())
```

Reply:
> So in that case, we'd want to apply the default connection limit setting to all torrents, including the ones that are currently uploading.

What default limit setting do you talk about? Is it about `m_maxConnectionsPerDownloadingTorrent`?
But aren't they now presented as two independent settings?